src/utils/device/dehydration.ts

```diff
@@ -15,43 +15,43 @@ limitations under the License.
 */
 
 import { logger } from "matrix-js-sdk/src/logger";
+import { CryptoApi } from "matrix-js-sdk/src/matrix";
 
 import { MatrixClientPeg } from "../../MatrixClientPeg";
 
-// the interval between creating dehydrated devices
-const DEHYDRATION_INTERVAL = 7 * 24 * 60 * 60 * 1000;
-
-// check if device dehydration is enabled
-export async function deviceDehydrationEnabled(): Promise<boolean> {
-    const crypto = MatrixClientPeg.safeGet().getCrypto();
+/**
+ * Check if device dehydration is enabled.
+ *
+ * Note that this doesn't necessarily mean that device dehydration has been initialised
+ * (yet) on this client; rather, it means that the server supports it, the crypto backend
+ * supports it, and the application configuration suggests that it *should* be
+ * initialised on this device.
+ *
+ * Dehydration can currently only enabled by setting a flag in the .well-known file.
```

Comment:
Nit (probably my fault, sorry):

```suggestion
 * Dehydration can currently only be enabled by setting a flag in the .well-known file.
```